rados/conn.go

```diff
@@ -59,6 +59,8 @@ func (c *Conn) Shutdown() {
 		return
 	}
 	C.rados_shutdown(c.cluster)
+	// set to nil, so that freeConn() does not call rados_shutdown() again.
+	c.cluster = nil
```

Comment:
This is a good change but I think we ought to make both the freeConn / Shutdown functions as idempotent as we can. It strikes me that both functions should behave similarly such that they share most of their behavior and/or code. (Call freeConn from within Shutdown?)
What do you think?

Reply:
Yes, that might be cleaner. I'll change it.